packages/taquito-utils/src/taquito-utils.ts

```diff
@@ -304,3 +305,56 @@ export function char2Bytes(str: string) {
 export function bytes2Char(hex: string): string {
   return Buffer.from(hex2buf(hex)).toString('utf8');
 }
+
+/**
+ *
+ * @description Convert hex string/UintArray/Buffer to bytes
+ *
+ * @param hex String/Uint8Array/Buffer value to convert to bytes
+ */
+export function hex2Bytes(hex: string | Uint8Array | Buffer): Buffer {
+  if (typeof hex === 'string' && hex.match(/[\da-f]{2}/gi)) {
```

Comment:
In the sapling state class, it seems like hex2Bytes is only called with a string as a parameter. I am curious to know why we need a function that supports Uint8Array and Buffer.

Reply:
Yeah I think keeping it as string only is okay. This helper function was present in airgap's implementation, and it had those param types. I think we can adapt this to fit our specific use cases, I will rework the function.